packages/pyright-internal/src/languageServerBase.ts

```diff
@@ -632,20 +632,22 @@ export abstract class LanguageServerBase implements LanguageServerInterface {
 
     protected onInitialized() {
         if (!this.client.hasWorkspaceFoldersCapability) {
-            // if folder capability is not supported, initialize ones given by onInitialize.
+            // If folder capability is not supported, initialize ones given by onInitialize.
             this.updateSettingsForAllWorkspaces();
             return;
         }
 
         this._connection.workspace.onDidChangeWorkspaceFolders((event) => {
-            event.removed.forEach((workspace) => {
-                const rootPath = this._uriParser.decodeTextDocumentUri(workspace.uri);
+            event.removed.forEach((workspaceInfo) => {
+                const rootPath = this._uriParser.decodeTextDocumentUri(workspaceInfo.uri);
+                const workspace = this._workspaceMap.getWorkspaceForFile(this, rootPath);
+                workspace.serviceInstance.dispose();
```

Comment:
Can we call `workspace.serviceInstance.dispose()` in `WorkspaceMap.delete` instead?

If not, there are two places where `_workspaceMap.delete` is called in Pylance for notebook scenarios, and they'll also need to get updated to dispose the `serviceInstance`.

Reply:
Yes, good call. I made the change.